[hardware_interface::RobotHW] doc: update read and write, fix: group names

fjp opened the pull request:
This PR is split from the large PR #433 and contains the following changes: 

- Update brief and detailes of read and write methods
- Fix doxygen group names and add new group name "Control Loop" to read and write methods.
  Removes these methods from group Hardware Interface Switching
- Remove too many details of class description in PR #433 -> new PR
- Remove too many details of `init` method in PR #433 -> new PR
- Apply suggestions from code review in #433 

It is related to [ros-controls/ros_controllers/#474](https://github.com/ros-controls/ros_controllers/pull/474).

Co-Authored-By: Matt Reynolds <mtreynolds@uwaterloo.ca>

fjp committed:
doc: update read and write, fix: group names

* update brief and detailes of read and write methods
* fix doxygen group names
* remove too many details of class description -> new PR
* remove too many details of init method -> new PR
* apply suggestions from code review
* add doxygen group name "Control Loop" to read and write methods.
  Removes these methods from group Hardware Interface Switching

Co-Authored-By: Matt Reynolds <mtreynolds@uwaterloo.ca>

fjp committed:
Merge remote-tracking branch 'upstream/melodic-devel' into extend-read-write-docstrings

fjp commented on hardware_interface/include/hardware_interface/robot_hw.h:
```diff
@@ -123,7 +124,9 @@ class RobotHW : public InterfaceManager
-/** \name Hardware Interface Switching
+  /**\}*/
+
+  /** \name Hardware Interface Switching
```
The [doxygen group](http://www.doxygen.nl/manual/grouping.html#memgroup) "Hardware Interface Switching" [looks different](https://github.com/ros-controls/ros_control/pull/433#issuecomment-604438698) than the others ("Control Loop" and "Resource Management"). This happens probably because of the `enum SwitchState` mentioned in [this discussion](https://github.com/ros-controls/ros_control/pull/433#discussion_r405916399). As the enum belongs to this group I would leave it this way even though the formatting looks different.

matthew-reynolds commented on hardware_interface/include/hardware_interface/robot_hw.h:
```diff
@@ -51,9 +51,9 @@ namespace hardware_interface
+ * the names of interface types derived from \ref HardwareInterface and
  * instances of those interface types.
- *
+ * 
```
Please remove trailing space. Same below on line 73.

matthew-reynolds commented on hardware_interface/include/hardware_interface/robot_hw.h:
```diff
@@ -80,6 +80,7 @@ class RobotHW : public InterfaceManager
    */
   virtual bool init(ros::NodeHandle& /*root_nh*/, ros::NodeHandle &/*robot_hw_nh*/) {return true;}
 
+
```
It doesn't look like the rest of the codebase is entirely consistent, but I think this extra newline should be removed to be in line with the majority.

matthew-reynolds commented on hardware_interface/include/hardware_interface/robot_hw.h:
```diff
@@ -148,33 +151,59 @@ class RobotHW : public InterfaceManager
   };
 
-  // Return (in realtime) the state of the last doSwitch()
+  /** \brief Return (in realtime) the state of the last doSwitch() */
```
Again, the codebase isn't entirely consistent on this, but I think these `\brief`s (Here and `switchResult()`) should end in a period.

matthew-reynolds commented on hardware_interface/include/hardware_interface/robot_hw.h:
```diff
@@ -148,33 +151,59 @@ class RobotHW : public InterfaceManager
+  /** \brief Write commands to the robot hardware.
+   * 
+   * The write method is part of the control loop cycle (\ref read, update, \ref write) 
+   * and used to send out commands to the robot's hardware 
```
```suggestion
   * and is used to send out commands to the robot's hardware 
```

matthew-reynolds reviewed (changes requested):
Thanks so much for separating this out. This looks good to me, here are a couple of nitpicks:

fjp committed:
Add sugestions from code review to robot_hw.h

- Remove whitespaces
- Remove newline
- Add periods to briefs
- Add "is" to docstring

Co-Authored-By: Matt Reynolds <mtreynolds@uwaterloo.ca>

fjp committed:
Update hardware_interface/include/hardware_interface/robot_hw.h

add "is" from suggestion

Co-Authored-By: Matt Reynolds <mtreynolds@uwaterloo.ca>

fjp committed:
Merge branch 'extend-read-write-docstrings' of https://github.com/fjp/ros_control into extend-read-write-docstrings

fjp committed:
remove trailing space

fjp commented:
Thanks @matthew-reynolds reviewing this. I've applied your suggestions, please let me know if something else needs to be fixed. After that I'll continue with the rest mentioned in #433.

bmagyar reviewed (approved):
Many thanks for this!
